modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -723,10 +731,12 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
         Halide::Func top = (name.empty() ? Halide::Func() : Halide::Func(name));
         Halide::RDom r(0, kernel.width, 0, kernel.height);
         Halide::Expr kx, ky;
-        if (pad.width || pad.height)
+        if(pad.left || pad.top || pad.right || pad.bottom)
         {
-            kx = clamp(x * stride.width + r.x - pad.width, 0, inWidth - 1);
-            ky = clamp(y * stride.height + r.y - pad.height, 0, inHeight - 1);
+            int max_pad_w = std::max(pad.left, pad.right);
+            int max_pad_h = std::max(pad.top, pad.bottom);
+            kx = clamp(x * stride.width + r.x - max_pad_w, 0, inWidth - 1);
+            ky = clamp(y * stride.height + r.y - max_pad_h, 0, inHeight - 1);
```

Comment:
Please fix this part of code. Replace `pad.left` to pad_l etc.
And why we use `max_pad_w` as a maximum between left and right paddings?

Reply:
fixed